### ISSUE TITLE
A simpler implmentation of device & car connectivity

### ISSUE THREAD

**Issue by n1snt** [organicmaps/organicmaps#4893]
**Is your feature request related to a problem? Please describe.**
Google maps on OnePlus devices (Idk if it's OnePlus specific because my other Pixel device seems to allow opening the app while connected to AA) currently is implemented in such a way that once the device is connected to Android Auto it closes the app & does not allow launching the app. It shows the following toast. (I think it calls the finish() method because I tried replicating the functionality on OM AA & was able to replicate it). 
I recently noticed this as I was trying out OM on the head unit & listing the bugs and creating the task for my proposal.

**Describe the solution you would like**
I liked this implementation because it's really simple & does not allow any errors from the user side (e.g selecting the wrong vehicle) & reduces distraction as well. 
Wouldnt this be a better implementation than what we have right now?
The flaws in the current implementation:
- Screen is always on and displays the following screen (More battery usage)
- Glitches? (SS given below)
- In future when we add navigation it would need a lot more error handling since the user could select the wrong vehicle.
- On top of that isn't the whole point of AA to not focus on the phone & instead use the head unit?
And this implementation fixes the issues given above & would remove lines of code rather than add more fixes.
 
**Additional context**
Google maps display this when connected to AA:
![image](https://user-images.githubusercontent.com/46300244/229184357-dd15f987-4800-4947-83c3-badbb8a92897.png)
This screen stays on when connected to AA even when locked:
![image](https://user-images.githubusercontent.com/46300244/229184549-78195943-7ae4-4b85-96c5-2c395fc8a82c.png)
Glitches:
![image](https://user-images.githubusercontent.com/46300244/229184701-2b926a8f-df4d-4725-90b0-8a6c2683a7c8.png)

I would love to have some feedback on this because I have gone through the app & have almost replicated google maps functionality.

**Comment by Jean-BaptisteC** [comment]
I think, the best way is to doing as similar of Apple CarPlay (I don't know, how it's works)
- Add fullscreen IHM with text is more easy to understand then little toast at the bottom of the screen
- Thanks for bugs with bottom sheet

CC @AndrewShkrob

**Comment by n1snt** [comment]
Let me do some research on Apple Carplay.
I will revert with my findings later.

**Comment by biodranik** [comment]
From the UX side, it is way easier for a driver to search/select the destination from the phone than from the CarPlay or AA unit. Google Maps and Apple Maps on iOS allow that. And that is VERY convenient on cars without touch screens. Even on touch screens, it's not easy to type some addresses, customize avoid options, to search, the phone is more suitable for that.

I assume that either there are some unknown technical limitations, or Google engineers are lazy to think about and implement a really good UX for AA.

**Comment by n1snt** [comment]
There are Android Auto head units without a touchscreen?

**Comment by biodranik** [comment]
Of course. There are MANY cars without touch screens. Or with a touch screen, but also with a convenient hand-control, when a driver can safely use his right hand without touching the screen and keeping his eyes on the road.

**Comment by n1snt** [comment]
I searched for a few but didn't find any head units with android auto but no touch screen.
I'm not sure how android auto would work without a touch screen.
If there are head units without touch screens then the Google maps AA implementation in my AA would be a better choice.

**Comment by AndrewShkrob** [comment]
@n1snt You can test how it works without a touchscreen in DHU
Use flag `--input=rotary` or `--input=hybrid`
Current AA implementation already supports it

Check this link: https://developer.android.com/training/cars/testing
There are a lot of interesting options that should also be supported (like cluster display, in-car sensors, voice input)

**Comment by n1snt** [comment]
Thanks a lot I was just going through the same resource :)

**Comment by n1snt** [comment]
How google maps has implemented apple carplay:

- Searching is available on both phone & head unit.
- Phone screen does not get blocked when accessing carplay.
- Maps are not connected i.e: when I move maps on phone it does not move the map on the head unit.
- When starting navigation on phone it automatically starts on the head unit.
- After navigation is started the phone only shows the directions & a navigation stop button.
- When navigation is started maps are not shown on the phone
- When permissions are not granted it works similarly to the AA version of google maps.
- Surprisingly when clicking on a location on the map it does nothing.

Key takeaways from this implementation:

- It is pretty similar to AA but the handling of navigation is a bit different, on carplay it starts displaying the destination even before starting the navigation but on AA it directly shows the navigation when navigation is started on the phone.
- We should keep the current implementation but should allow accessing both the phone & head unit at the same time.
- Is there any way to render 2 maps? One for the head unit & one for the phone. @AndrewShkrob 
- Allowing the phone screen to turn off & do not show AA on the lock screen when connected to AA.

Some screenshots:
Before starting navigation:
![image](https://user-images.githubusercontent.com/46300244/229369965-695b0e94-c04b-4738-9ab9-782b978f36bd.png)

After starting navigation
![image](https://user-images.githubusercontent.com/46300244/229370041-bd093a3e-4184-48cc-9d40-af4695119524.png)

**Comment by AndrewShkrob** [comment]
> Is there any way to render 2 maps? One for the head unit & one for the phone.

It is possible but it requires a lot of work in the core. Check this: https://github.com/organicmaps/organicmaps/pull/4080
We plan to finish with this implementation when we can freely swap between the device and AA and it's possible to search, build a route on the phone then swap to AA to use navigation mode.
Then we will work on rendering to support many surfaces.

**Comment by n1snt** [comment]
[This](https://github.com/organicmaps/organicmaps/pull/4080) looks promising :100: 
While this is being worked on I can get the permissions, directions & other stuff working:)

**Comment by rtsisyk** [comment]
@n1snt the short-term plan is to launch with "Continue on phone" / "Continue in car" implementation, like on your screenshots. Other ideas can be tested later. Please try out [aa](https://github.com/organicmaps/organicmaps/tree/aa) branch.
